extensions/amp-subscriptions/0.1/amp-subscriptions.js

```diff
@@ -294,8 +295,11 @@ export class SubscriptionService {
             this.fetchEntitlements_(subscriptionPlatform);
           }
       );
-      this.startAuthorizationFlow_();
 
+      isStoryDocument(this.ampdoc_).then(isStory => {
+        // Delegates the platform selection and activation call if is story.
+        this.startAuthorizationFlow_(!isStory /** doPlatformSelection */);
```

Comment:
So, in case of story this method is eventually called? From where and how?

Reply:
It’d work this way:
1. (This line of code) amp-subscriptions starts, processes the grant state, and delays the selectAndActivatePlatform if it detects that it is in an amp-story context
2. User navigates until they hit a non-free page, and amp-story calls a new method on the SubscriptionsService to trigger `maybeSelectAndActivatePlatform` (right below), which could activate the local platform (dialog), or the SwG UI, etc, but still pays attention to not overriding the viewer provided auth, or `alwaysGrant` option.

The design doc has a bit more information if you want :) Does it make sense?